test/e2e/storage/flexvolume.go

```diff
@@ -0,0 +1,207 @@
+	gciVolumePluginDir = "/etc/srv/kubernetes/kubelet-plugins/volume/exec"
+)
+
+// testFlexVolume tests that a client pod using a given flexvolume driver
+// successfully mounts it and runs
+func testFlexVolume(driver string, cs clientset.Interface, config framework.VolumeTestConfig, f *framework.Framework, clean bool) {
+	tests := []framework.VolumeTest{
+		{
+			Volume: v1.VolumeSource{
+				FlexVolume: &v1.FlexVolumeSource{
+					Driver: "k8s/" + driver,
+				},
+			},
+			File: "index.html",
+			// Must match content of examples/volumes/flexvolume/dummy(-attachable) domount
+			ExpectedContent: "Hello from flexvolume!",
+		},
+	}
+	framework.TestVolumeClient(cs, config, nil, tests)
+
+	if clean {
+		framework.VolumeTestCleanup(f, config)
+	}
+}
+
+// installFlex installs the driver found at filePath on the node and restarts
+// kubelet. If node is nil, installs on the master and restarts
+// controller-manager.
+func installFlex(node *v1.Node, vendor, driver, filePath string) {
+	flexDir := getFlexDir(node == nil, vendor, driver)
+	flexFile := path.Join(flexDir, driver)
+
+	host := ""
+	if node != nil {
+		host = framework.GetNodeExternalIP(node)
+	} else {
+		host = net.JoinHostPort(framework.GetMasterHost(), sshPort)
+	}
+
+	cmd := fmt.Sprintf("sudo mkdir -p %s", flexDir)
+	sshAndLog(cmd, host)
+
+	data := generated.ReadOrDie(filePath)
+	cmd = fmt.Sprintf("sudo tee <<'EOF' %s\n%s\nEOF", flexFile, string(data))
+	sshAndLog(cmd, host)
+
+	cmd = fmt.Sprintf("sudo chmod +x %s", flexFile)
+	sshAndLog(cmd, host)
+
+	if node != nil {
+		err := framework.RestartKubelet(host)
+		framework.ExpectNoError(err)
+		err = framework.WaitForKubeletUp(host)
+		framework.ExpectNoError(err)
+	} else {
+		err := framework.RestartControllerManager()
+		framework.ExpectNoError(err)
+		err = framework.WaitForControllerManagerUp()
+		framework.ExpectNoError(err)
+	}
+}
+
+func uninstallFlex(node *v1.Node, vendor, driver string) {
+	flexDir := getFlexDir(node == nil, vendor, driver)
+
+	host := ""
+	if node != nil {
+		host = framework.GetNodeExternalIP(node)
+	} else {
+		host = net.JoinHostPort(framework.GetMasterHost(), sshPort)
+	}
+
+	cmd := fmt.Sprintf("sudo rm -r %s", flexDir)
+	sshAndLog(cmd, host)
+}
+
+func getFlexDir(master bool, vendor, driver string) string {
+	volumePluginDir := defaultVolumePluginDir
+	if framework.ProviderIs("gce") {
+		if (master && framework.MasterOSDistroIs("gci")) || (!master && framework.NodeOSDistroIs("gci")) {
+			volumePluginDir = gciVolumePluginDir
+		}
+	}
+	flexDir := path.Join(volumePluginDir, fmt.Sprintf("/%s~%s/", vendor, driver))
+	return flexDir
+}
+
+func sshAndLog(cmd, host string) {
+	result, err := framework.SSH(cmd, host, framework.TestContext.Provider)
+	framework.LogSSHResult(result)
+	framework.ExpectNoError(err)
+	if result.Code != 0 {
+		framework.Failf("%s returned non-zero, stderr: %s", cmd, result.Stderr)
+	}
+}
+
+var _ = framework.KubeDescribe("Flexvolumes [Volume][Disruptive]", func() {
```

Comment:
You can use SIGDescribe, instead of KubeDescribe.  It will automatically add the [sig-storage] tag to the test name.